### PR TITLE
refactor(pie-cookie-banner): DSW-2207 refactored experimental package from cookie banner using page object modal

### PR DESCRIPTION
**Pull request by JoshuaNg2332**
## Describe your changes (can list changeset entries if preferable)
- Refactored `@sand4rt/experimental-ct-web` out of `component` and `accessibility` tests
- Updated `base.page.ts` file to include a new load function for component objects to use
- Updated Cookie Banner page object to support new load function

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If changes will affect consumers of the package, I have created a changeset entry.

### Reviewer 1 - @xander-marjoram 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview

### Reviewer 2
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview

### CODE REVIEW

File: packages/components/pie-webc-testing/src/helpers/page-object/base-page.ts [hunk 1]
```diff
@@ -1,10 +1,86 @@
...
+
+        return `&args=${flattenQueries(queries).join(';')}`;
+    }
+
+    /**
+     * Listens for the given event. When it is emitted, the name of the event is pushed to an array on the `window` object.
+     */
+    async listenForEvent (eventName: string) {
+        await this.page.evaluate(() => {
+            window.__eventsArray = [];
```
Comment (xander-marjoram) [inline packages/components/pie-webc-testing/src/helpers/page-object/base-page.ts:65]: Just noticed that a side effect of this is that it will clear out any events that are already in the array. I don't think that's an issue if the setup is all done at the start of the test but just something to bear in mind.

File: packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts [hunk 2]
```diff
@@ -177,11 +143,9 @@ test.describe('PieCookieBanner - Component tests', () => {
...
-    test('should toggle all preferences if the `all` preference node is set to true', async ({ mount }) => {
+    test('should toggle all preferences if the `all` preference node is set to true', async () => {
         // Arrange
-        await mount(PieCookieBanner, {
-            props: {} as CookieBannerProps,
-        });
+        await pieCookieBannerComponent.load({ defaultPreferences: null });
 
         // Act
         await pieCookieBannerComponent.clickManagePreferencesAction();
```
Comment (xander-marjoram) [inline packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts:151]: Candidate for a future PR, do you think it would be worth wrapping this logic that gets all the preferences?

File: packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts [hunk 3]
```diff
@@ -309,16 +273,9 @@ test.describe('PieCookieBanner - Component tests', () => {
...
-                {
-                    props: {
-                        hasPrimaryActionsOnly: true,
-                    },
-                },
-            );
+            await pieCookieBannerComponent.load({ hasPrimaryActionsOnly: true });
 
             // Act
             const acceptAllButtonVariant = await pieCookieBannerComponent.getAcceptAllVariant('variant');
```
Comment (xander-marjoram) [inline packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts:281]: Not part of this PR, but the naming of this looks a bit off `getAcceptAllVariant('variant')` sounds like it has some redundancy in there.

File: packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts [hunk 4]
```diff
@@ -352,14 +304,9 @@ test.describe('PieCookieBanner - Component tests', () => {
...
-            test('should set a default cookie statement link of empty string within the banner description', async ({ mount }) => {
+            test('should set a default cookie statement link of empty string within the banner description', async () => {
                 // Arrange & Act
-                await mount(
-                    PieCookieBanner,
-                    {
-                        props: {},
-                    },
-                );
+                await pieCookieBannerComponent.load({ cookieStatementLink: '' });
```
Comment (xander-marjoram) [inline packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts:309]: Is the output of this still the same without passing the prop value in? The test reads like it should be testing the default value rather than what happens if you pass in an empty string.

**Review by xander-marjoram — DISMISSED** [review]

File: packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts (hunk 4 above)
Comment (JoshuaNg2332) [reply packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts:309]: No, this then defaults back. We are testing the behaviour of passing an empty string into the link.

File: configs/pie-components-config/playwright-lit-config.js [hunk 5]
```diff
@@ -52,5 +52,11 @@ export function getPlaywrightConfig () {
                 testMatch: ['**/test/accessibility/*.spec.{js,ts}'],
             },
         ],
+        webServer: !process.env.CI ? {
+            command: 'npx turbo dev --filter=pie-storybook',
```
Comment (siggerzz) [inline configs/pie-components-config/playwright-lit-config.js:56]:
```suggestion
        webServer: !process.env.CI ? {
            command: './node_modules/.bin/turbo dev --filter=pie-storybook',
```

**Review by siggerzz — COMMENTED** [review]
Small comment, also yarn.lock has a conflict!

File: configs/pie-components-config/playwright-lit-config.js (hunk 5 above)
Comment (JoshuaNg2332) [reply configs/pie-components-config/playwright-lit-config.js:56]: This line doesn't work unfortunately.

**Review by siggerzz — APPROVED** [review]
